tool/tsh/tsh.go

```diff
@@ -4736,3 +4742,51 @@ func onHeadlessApprove(cf *CLIConf) error {
 	})
 	return trace.Wrap(err)
 }
+
+var mlockModes = []string{mlockModeNo, mlockModeAuto, mlockModeBestEffort, mlockModeStrict}
+
+const (
+	// mlockModeNo disables locking process memory.
+	mlockModeNo = "off"
+	// mlockModeAuto automatically chooses whether memory locking will be attempted and/or enforced.
+	mlockModeAuto = "auto"
+	// mlockBestEfforts enables locking process memory, but errors will be ignored and logged.
+	mlockModeBestEffort = "best_effort"
+	// mlockModeStrict enables locking process memory and enforces it succeeds without errors.
+	mlockModeStrict = "strict"
+
+	// mlockFailureMessage is a user readable message for mlock errors and debug logs.
+	mlockFailureMessage = "Failed to lock process memory for headless login. " +
+		"Memory locking is used to prevent secrets in memory from being swapped to disk. " +
+		"Please ensure that memory locking is available on your system and your user has " +
+		"locking privileges. This means using a Linux operating system and increasing " +
+		`your user's memory locking limit to unlimited if needed. Alternatively, ` +
+		"set --mlock=off to disable it. This is not recommended in production " +
```

Comment:
Should we mention environment variable in the message as well, in addition to the CLI flag?